### PR TITLE
Allow subprojects to override the icons used for clothing

### PR DESCRIPTION
**Pull request by nullquery**
In my case I'm including the tgstation.dme file from my own project, but I'd like to use custom clothing items. Presently the code does not allow me to override the icon used when rendering the icon on top of the player (apart from the left/right hand states).

This change allows the icon to be modified by setting the item_icon variable on the item.

### CODE REVIEW

**Comment by Jordie0608** [comment]
>6 commits
>3 lines changed

**Comment by nullquery** [comment]
I don't know. I get those "merge pull request" commits automatically. Am I doing something wrong?

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 1]
```diff
@@ -457,7 +457,7 @@ generate/load female uniform sprites matching all previously decided variables
 	if(!isinhands && alternate_worn_icon)
 		file2use = alternate_worn_icon
 	if(!file2use)
-		file2use = default_icon_file
+		file2use = item_icon || default_icon_file
```
Comment (duncathan) [inline code/modules/mob/living/carbon/human/update_icons.dm]: Woah, does this work as a null coalescing operation?

Comment (nullquery) [inline code/modules/mob/living/carbon/human/update_icons.dm]: Yup. If item_icon is null then it falls back on default_icon_file.

**Comment by nullquery** [comment]
Obligatory BYOND bug regarding files in subprojects: http://www.byond.com/forum/?post=2017307

I'd still like to see this pulled regardless because ideally I only want to provide my own icons (and not have to edit /tg/ code every time I merge my code with /tg/)

**Comment by GinjaNinja32** [comment]
>I don't know. I get those "merge pull request" commits automatically. Am I doing something wrong?

You're updating your fork via the GitHub web UI/PRs, you need to do it from the command-line or whatever Git client you're using. Pull from tgstation/-tg-station master, then push to nullstation/-tg-station master.

**Comment by RemieRichards** [comment]
Why not just use ```alternate_worn_icon```?

**Comment by nullquery** [comment]
alternate_worn_icon doesn't seem to be consistently applied to every type, and I'm not sure if this is its intended use case.

While writing the code I did think that alternate_worn_icon could be deprecated in favor of item_icon, but I honestly didn't think about who was using it at the time.

**Comment by RemieRichards** [comment]
@nullquery not consistently applied to every type? I'm not 100% sure what you're meaning here but it's defined in the same place your new variable is, and performs all of the functions of your new variable, thus making your new variable redundant.

**Comment by nullquery** [comment]
Oh, right. I was looking at an older version of the code at the time...

Well, if there's already a variable that does this then I'll revert my changes once I get back from work.

Why was the name "alternate_worn_icon" chosen instead of "item_icon"? Is there a use case of" item_state" which I'm not aware of?

**Comment by RemieRichards** [comment]
'Alternate' was chosen as it's an alternate value, if it's there it's used but otherwise it is unused, this was done to avoid having to change ALL the use cases of icon and default icons to use a new variable, and due to my original objective of adding the alternate var for admins to mess around with. Plus "item" is a misnomer, not all worn objects draw from ```item_state``` for how they're worn (All do for how they're <i>held</i> though), some use ```icon_state```, with icon obviously already taken as a byond var AND used for on ground/inventory states.

tl;dr the naming scheme fits purpose

**Comment by nullquery** [comment]
OK, thanks for the info!! I'll look at this after work.
